service/src/main/java/bio/terra/pipelines/common/utils/PipelineInputTypesEnum.java

```diff
@@ -0,0 +1,96 @@
+
+public enum PipelineInputTypesEnum {
+  STRING {
+    @Override
+    public String cast(String fieldName, Object value) {
+      validateNotNullOrEmpty(fieldName, value);
+      String stringValue = objectMapper.convertValue(value, String.class);
+      if (stringValue.isBlank()) {
+        throw new ValidationException(String.format("%s must not be empty", fieldName));
+      }
+      return stringValue;
+    }
+  },
+  INTEGER {
+    @Override
+    public Integer cast(String fieldName, Object value) {
+      validateNotNullOrEmpty(fieldName, value);
+      // objectMapper will accept a float and convert it to an integer; we don't want this
+      if (value instanceof Float || value instanceof Double) {
+        throw new IllegalArgumentException(
+            "Integer input must be an integer"); // this error message gets overwritten
+      }
+      return objectMapper.convertValue(value, Integer.class);
+    }
+  },
+  VCF {
+    @Override
+    public String cast(String fieldName, Object value) {
+      validateNotNullOrEmpty(fieldName, value);
+      String stringValue = objectMapper.convertValue(value, String.class);
+      if (!stringValue.endsWith(".vcf.gz")) {
+        throw new ValidationException(
+            String.format("%s must be a path to a VCF file ending in .vcf.gz", fieldName));
+      }
+      return stringValue;
+    }
+  },
+  ARRAY_STRING {
+    @Override
+    public String[] cast(String fieldName, Object value) {
+      validateNotNullOrEmpty(fieldName, value);
+      String[] stringArray = objectMapper.convertValue(value, String[].class);
+      validateNotEmptyArray(fieldName, stringArray);
+      for (String stringValue : stringArray) {
+        try {
+          STRING.cast(fieldName, stringValue);
+        } catch (ValidationException e) {
+          throw new ValidationException(
+              String.format("%s must be an array of non-null strings", fieldName));
+        }
+      }
+      return stringArray;
+    }
+  },
+  ARRAY_VCF {
+    @Override
+    public String[] cast(String fieldName, Object value) {
+      validateNotNullOrEmpty(fieldName, value);
+      String[] stringArray = objectMapper.convertValue(value, String[].class);
+      validateNotEmptyArray(fieldName, stringArray);
+      for (String stringValue : stringArray) {
+        try {
+          VCF.cast(fieldName, stringValue);
+        } catch (ValidationException e) {
+          throw new ValidationException(
+              String.format(
+                  "%s must be an array of non-null paths to VCF files ending in .vcf.gz",
+                  fieldName));
+        }
+      }
+      return stringArray;
+    }
+  };
+  final ObjectMapper objectMapper = new ObjectMapper();
+
+  public abstract <T> T cast(String fieldName, Object value);
+
+  private static void validateNotNullOrEmpty(String fieldName, Object value) {
+    if (value == null) {
+      throw new ValidationException(String.format("%s must not be null", fieldName));
```

Comment:
If this is java 17+, you can do:
```suggestion
      throw new ValidationException("%s must not be null".format(fieldName));
```